Security page revision

GBKS opened the pull request:
Implementing suggested changes in #656. This is a slight revision, tweaking copy details and visuals. Did not add anything new really.

- Revised visuals for consistency (like keypad-based home screen)
- Note about first-use flow to set up users for good security
- Note about spaced repetition
- Added full-screen modal with a specific security reminder
- Removed privacy checklist (thinking a security checklist might be enough)
- Added big images for display in modals
- Note about using PIN to approve certain actions
- Note about Blixt and deleting the recovery phrase
- Did not touch the watchtowers section

I wonder if we need to add another page to this reference design that solely shows off settings. We have quite a few different settings screens (general, privacy & security, backup, help, "advanced", addresses, fees) but they are spread over various pages.

Keeping this as a draft for now as I'd like to review again with a fresh brain tomorrow before opening it up for wider review.

🤜[Man, what a preview](https://deploy-preview-872--bitcoin-design-site.netlify.app/guide/daily-spending-wallet/security/)🤛

Closes #656

Bosch-0 commented on guide/daily-spending-wallet/security.md:
```diff
@@ -166,39 +158,21 @@ Bitcoin wallet users may feel hesitant about the privacy implications of your pr
 
 ### PIN protection
```
Passphrases are generally stronger than PINs, we should mention those in this "Prevent unwanted access" section.

Bosch-0 commented on guide/daily-spending-wallet/security.md:
```diff
@@ -166,39 +158,21 @@ Bitcoin wallet users may feel hesitant about the privacy implications of your pr
+Alternative methods to PIN protection include passwords, as well as swipe patterns.
 
 {% include tip/close.html %}
 
```
The idea of a "Brick me PIN" or wipe PIN that essentially wipes the users wallet clean (they would still have their backups hopefully) is something else worth exploring on this page.

GBKS replied:
Do you know of any wallets that do this?

Bosch-0 replied:
Zeus ;) haha

GBKS replied:
Added a note on this in the newly added "Delete wallet" paragraph.

Bosch-0 commented on guide/daily-spending-wallet/security.md:
```diff
@@ -226,29 +201,7 @@ There may be situations in which users do not want to show their real balances a
 
 To further prevent unwanted spending, wallet limits can be established. These can limit the size of single transactions, or impose daily limits. Limits should be changeable at any time, but they should only apply after a certain amount of time has passed. Alternatively, a dedicated password could be required to change limits instantly. So even if a third-party manages to access the wallet, the pa ...
 
```
The grey background used in the Privacy & Security settings image in Wallet limits section is inconsistent with other frames in the UI kit / has bad contrast with the background of the gallery, could make it white or a lighter grey?

Bosch-0 replied:
This also applies to several images in below this one in watchtowers and deleting the recovery phrase sections

Bosch-0 commented on guide/daily-spending-wallet/security.md:
```diff
@@ -278,7 +232,7 @@ However, this prevents mobile lightning wallets from being reliable monitors of
+Another technique is to use [watchtowers](https://wiki.ion.radar.tech/tech/research/watchtowers). These are third parties that continuously monitor wallets and punish bad actors who attempt to cheat the wallet they are monitoring. Watchtowers should be provided by a different party than the wallet and payment channels, to reduce the risk of collusion.
 
 {% include image-gallery.html pages = page.images_watchtower %}
 
```
Good review overall. Some suggestions of stuff to add:

- Make it so users can't screenshot recovery phrases when manually backing up (or other sensitive info)
- Scrambled PIN
- Option for users to securely erase wallet (say if they are moving from one phone to another)

GBKS replied:
What do you mean by scrambled PIN?

Bosch-0 replied:
Scrambles / randomizes the layout of the PIN numbers. If a camera or someone's sees your pattern of entry they can quite easily know the PIN entered. 

Example in Samourai below. Zeus also offers this

![image](https://user-images.githubusercontent.com/55287964/179701411-e7fe6ac8-615a-4a73-83a3-e0ee10cf8a0c.png)

GBKS replied:
Not being able to screenshot the recovery phrase is a good addition to the manual backup page.

I just added scrambled PINs and erasing the wallet to the page.

Bosch-0 commented:
Another topic we could touch on is timelocks. Some wallets, such as Citadel and Green wallet, now can lock users funds for X amount time for security reasons. Though this may be more relevant in the savings wallet

GBKS commented:
@Bosch-0 I think I addressed your feedback.

As for timelocks, that is more appropriate for an on-chain wallet. The daily spending wallet has wallet limits, which serve a similar purpose (slowing down how quickly money can be transferred out).

sbddesign commented on guide/daily-spending-wallet/security.md:
```diff
@@ -64,89 +111,38 @@ Bitcoin wallets contain sensitive information such as [recovery phrases]({{ '/gu
 ## Proactive support
 
-"Usable security & privacy" has become its own topic because it is not enough to simply implement features, it is just as important that users can easily and effectively use them. A highly effective security setting is of no use if users do not realize it exists. So applications should do their best to guide and inform users to adapt appropriate habits and practices, based on where users are in t ...
+"Usable security & privacy" has become a topic of its own because it is not enough to simply implement features, it is just as important that users can easily and effectively use them. A highly effective security setting is of no use if users do not realize it exists. So applications should do their best to guide and inform users to adapt appropriate habits and practices, based on where users are ...
```
```suggestion
"Usable security & privacy" has become a topic of its own because it is not enough to simply implement features. It is just as important that users can easily and effectively use them. A highly effective security setting is of no use if users do not realize it exists. So applications should do their best to guide and inform users to adapt appropriate habits and practices, based on where users are in the [usage life cycle]({{ '/guide/designing-products/usage-life-cycle/' | relative_url }}), as well as the application use case.
```

sbddesign commented on guide/daily-spending-wallet/security.md:
```diff
@@ -166,39 +162,21 @@ Bitcoin wallet users may feel hesitant about the privacy implications of your pr
-A PIN is one of the most common and traditional methods for securing a mobile application. This setup requires that users create a 4-6 digit PIN that they must enter when opening their wallet application or viewing sensitive information, such as their recovery phrase.
-
-<div class="image-slide-gallery">
+A [PIN](https://en.wikipedia.org/wiki/Personal_identification_number) (Personal identification number) is one of the most common and traditional methods for securing a mobile application. This setup requires users to choose a 4-6 digit number. They must enter it when opening their wallet application, or viewing sensitive information, such as their recovery phrase.
```
```suggestion
A [PIN](https://en.wikipedia.org/wiki/Personal_identification_number) (personal identification number) is one of the most common and traditional methods for securing a mobile application. This setup requires users to choose a 4-6 digit number. They must enter it when opening their wallet application, or viewing sensitive information, such as their recovery phrase.
```

Weird to capitalize just the word "personal". Need to pick a lane. Start each word with lowercase. Alternatively, title-case it to draw attention to the fact that the words form the acronym.

GBKS replied:
Updated.

sbddesign commented on guide/daily-spending-wallet/security.md:
```diff
@@ -10,15 +10,62 @@ redirect_from:
+      modalImage: choose-pin-big
+      alt: Choosing a pin screen
+      caption: When choosing a PIN, the user is also informed about how it is used.
+    - file: enter-pin
+      modalImage: enter-pin-big
+      alt: Enter pin screen
+      caption: PIN entry could be limited to a certain amount of unsuccessful attempts per day.
+    - file: enter-pin-scrambled
+      modalImage: enter-pin-scrambled-big
+      alt: Enter pin screen with scrambled numbers
+      caption: Scrambling numbers can make it harder for bystanders to see what was typed.
```
`enter-pin` and `enter-pin-scrambled` both appear to be the exact same image. An error?

GBKS replied:
Different is that the keypad numbers are scrambled (not `1-2-3...` but `3-7-8...`).

sbddesign commented on guide/daily-spending-wallet/security.md:
```diff
@@ -10,15 +10,62 @@ redirect_from:
+      alt: Wallet home screen with an overlay explaining that the daily spending limit was reached
+      caption: Users are notified when they reach the wallet limits.
 images_watchtower:
     - file: lightning-network-settings
```
Should the "sync notifications" be turned on by default for the user? 🚴

GBKS replied:
Good one. Updated.

sbddesign reviewed (changes requested):
Looking really good. I pointed out some typographical things and flagged a potential issue with one of the images.

GBKS commented:
@sbddesign and @Bosch-0, thanks for the feedback. I think I addressed it all and the page is ready for another review.

Bosch-0 reviewed (approved):
LGTM !

sbddesign reviewed (approved):
🪁
